File: app/Models/Store.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Store extends Model
{
    use HasFactory;

    protected $fillable = [
        'name', 'address', 'latitude', 'longitude', 'contact_phone', 'contact_name', 'is_active',
    ];

    protected function casts(): array
    {
        return [
            'latitude' => 'decimal:8',
            'longitude' => 'decimal:8',
            'is_active' => 'boolean',
        ];
    }

    public function schedules()
    {
        return $this->hasMany(Schedule::class);
    }

    public function checkIns()
    {
        return $this->hasMany(CheckIn::class);
    }

    public function sopChecklists()
    {
        return $this->hasMany(SopChecklist::class);
    }
}

File: mobile/lib/models/store.dart
class StoreModel {
  final int id;
  final String name;
  final String address;
  final double latitude;
  final double longitude;
  final String? contactPhone;
  final String? contactName;

  StoreModel({
    required this.id,
    required this.name,
    required this.address,
    required this.latitude,
    required this.longitude,
    this.contactPhone,
    this.contactName,
  });

  factory StoreModel.fromJson(Map<String, dynamic> json) {
    return StoreModel(
      id: json['id'],
      name: json['name'],
      address: json['address'],
      latitude: double.parse(json['latitude'].toString()),
      longitude: double.parse(json['longitude'].toString()),
      contactPhone: json['contact_phone'],
      contactName: json['contact_name'],
    );
  }
}
